### ISSUE TITLE
Incorrect call to PerformanceObserverWin on Linux platforms running under .net core

### ISSUE THREAD

**Issue by o-mdr** [espertechinc/nesper#24]
Related to #22, where an example source code can be taken. Run it on Linux (given you have applied a workaround for .net core serialisation of timezones https://github.com/espertechinc/nesper/pull/23). 

It looks like `PerformanceObserverWin` class is resolved incorrectly to trace performance counters on Linux platform, the class tries to load a Windows dll that doesn't exist on Linux. The following exception is thrown while NEsper engine is being initialised.

```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'com.espertech.esper.compat.threading.SlimLock' threw an exception. ---> System.TypeInitializationException: The type initializer for 'com.espertech.esper.compat.PerformanceObserverWin' threw an exception. ---> System.DllNotFoundException: Unable to load shared library 'Kernel32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libKernel32.dll: cannot open shared object file: No such file or directory
   at com.espertech.esper.compat.PerformanceObserverWin.QueryPerformanceFrequency(Int64& lpFrequency)
   at com.espertech.esper.compat.PerformanceObserverWin.Calibrate()
   at com.espertech.esper.compat.PerformanceObserverWin..cctor()
   --- End of inner exception stack trace ---
   at com.espertech.esper.compat.threading.SlimLock..cctor()
   --- End of inner exception stack trace ---
   at com.espertech.esper.compat.threading.SlimLock..ctor()
   at com.espertech.esper.compat.threading.MonitorSlimLock..ctor(Int32 lockTimeout)
   at com.espertech.esper.compat.collections.CopyOnWriteList`1..ctor()
   at com.espertech.esper.compat.collections.CopyOnWriteArraySet`1..ctor()
   at com.espertech.esper.filter.FilterServiceBase..ctor(ILockManager lockManager, FilterServiceGranularLockFactory lockFactory, Boolean allowIsolation)
   at com.espertech.esper.filter.FilterServiceLockCoarse..ctor(ILockManager lockManager, IReaderWriterLockManager rwLockManager, Boolean allowIsolation)
   at com.espertech.esper.filter.FilterServiceProvider.NewService(ILockManager lockManager, IReaderWriterLockManager rwLockManager, FilterServiceProfile filterServiceProfile, Boolean allowIsolation)
   at com.espertech.esper.core.service.EPServicesContextFactoryDefault.CreateServicesContext(IContainer container, EPServiceProvider epServiceProvider, ConfigurationInformation configSnapshot)
   at com.espertech.esper.core.service.EPServiceProviderImpl.DoInitialize(Nullable`1 startTime)
   at com.espertech.esper.core.service.EPServiceProviderImpl..ctor(IContainer container, Configuration configuration, String engineURI, IDictionary`2 runtimes)
   at com.espertech.esper.client.EPServiceProviderManager.GetProviderInternal(IContainer container, Configuration configuration, String providerURINonNull)
   at com.espertech.esper.client.EPServiceProviderManager.GetProvider(IContainer container, String providerURI, Configuration configuration)
   at com.espertech.esper.client.EPServiceProviderManager.GetDefaultProvider(IContainer container)
   at com.espertech.esper.client.EPServiceProviderManager.GetDefaultProvider()
   %user_callstack_starts_here%
```

**Comment by ajaxx** [comment]
Interesting, I'll see if I can reproduce this under NEsper 8.

**Comment by ajaxx** [comment]
Sorry, I didn't get this into NEsper 8.0.0.  However, I am looking into this right now and will fix it in NEsper 8.5.0 and will issue a backpatch.  The issue is that the preprocessor assumes mono or nothing else.  This goes back some time ago and we need a more suitable solution for .NET core.  I will see if there is a better answer.

**Comment by ajaxx** [comment]
NEsper 8.5.0 is finalized and has much better support for .NET core.  It will build and produce nupkg outputs for .netcoreapp 3.1.  Would be interested in hearing if you have any further issues with this.
